react/SelectBox/SelectBoxWithFixedOptions.js

```diff
@@ -0,0 +1,86 @@
+import { components } from 'react-select'
+import React from 'react'
+import groupBy from 'lodash/groupBy'
+
+import SelectBox from './SelectBox'
+import { silver } from '../../stylus/settings/palette.json'
+
+const optionPadding = 0.25
+
+const fixedOptionsStyle = {
+  paddingTop: optionPadding + 'rem',
+  paddingBottom: optionPadding + 'rem',
+  borderTop: `1px solid ${silver}`,
+  background: 'white',
+  position: 'relative'
+}
+
+const FixedOptions = props => (
+  <div style={fixedOptionsStyle}>{props.children}</div>
+)
+
+/*
+  This is the hacky part, where fixed options vnodes are manually spliced
+  from the options vnodes. Fixed options vnodes are returned along with
+  the menuUi where fixed vnodes have been removed.
+
+  We rely on react-select internal structure `menu > scroll captor > menu list > options`.
+  This code should be updated if react-select's internal structure changes.
+
+  https://github.com/cozy/cozy-ui/pull/501
+*/
+const extractFixed = children => {
+  const newChildren = React.cloneElement(children)
+  const menuList = newChildren.children ? newChildren.children[0] : null
+  if (!menuList) {
+    return { fixed: [], newChildren }
+  }
+  const options = menuList.children
+  const { true: fixed, false: normal } = groupBy(
+    options,
+    vnode => vnode.attributes.data.fixed === true
+  )
+  newChildren.children[0].children = normal
+  return { fixed, newChildren }
```

Comment:
Just my personal point but I think this part is not readable. We have to understand that you will get an object with two properties `true` and `false` (not usual as property names) to become then the `fixed` and `normal` objects.

I would (personally) prefer a classical `map() usage` here:
```javascript
const fixed = []
const normal = []
options.map(vnode => {
  if (vnode.attributes.data.fixed === true) { fixed.push(vnode) }
  else { normal.push(vnode) }
})
newChildren.children[0].children = normal
return { fixed, newChildren }
```

But this point is not blocking since it does the job.

Reply:
More a `forEach` than a `map`, even a `reduce`

Reply:
I changed to 

```
```patch
-  const { true: fixed, false: normal } = groupBy(
+  const { fixed, normal } = groupBy(
     options,
-    vnode => vnode.attributes.data.fixed === true
+    vnode => (vnode.attributes.data.fixed === true ? 'fixed' : 'normal')
   )
```

I do not find `reduce` appropriate. Sure, eventually you are reducing the array to an object, but it seems like an implementation detail. The intention here is to group the elements via a predicate and this is what groupBy does and imply.